drivers/media/i2c/ov5647.c

```diff
@@ -1140,14 +1171,16 @@ static int ov5647_enum_frame_size(struct v4l2_subdev *sd,
 {
 	struct ov5647_mode *mode = NULL;
 
-	if (fse->code == MEDIA_BUS_FMT_SBGGR8_1X8) {
+	if (fse->code == ov5647_get_mbus_code(sd, 1)) {
 		if (fse->index >= ARRAY_SIZE(supported_modes_8bit))
 			return -EINVAL;
 		mode = &supported_modes_8bit[fse->index];
-	} else if (fse->code == MEDIA_BUS_FMT_SBGGR10_1X10) {
+		fse->code = ov5647_get_mbus_code(sd, 1);
```

Comment:
Is this not a no-op?
```
    if (a == func()) {
        a = func();
```
looks wrong.

Reply:
Yes actually, that would seem to be pointless! Let me fix it up...